src/Payment.ts

```diff
@@ -527,3 +528,16 @@ function encode({
     return '0x'
   }
 }
+
+function processExtraData(transferDetails): TransferDetails {
```

Comment:
This is basically 99% the same as we already have in `Events.ts`. They are called the same, basically do the same and both were affected by this PR in the same manner (hint that the functionality is actually tight). Could we make this a single function and maybe put it as utility to `extraData.ts`?

Reply:
makes sense